diff --git a/dist/esm/staticFunctionMiddleware.js b/dist/esm/staticFunctionMiddleware.js
index c6fecf9ebb3ce93e574642f49bf66495bc62d9a0..de31551c0251f0e43859d8da83c5790ad3a5e56e 100644
--- a/dist/esm/staticFunctionMiddleware.js
+++ b/dist/esm/staticFunctionMiddleware.js
@@ -56,12 +56,22 @@ var fetchItem = async ({ data, functionId }) => {
 		functionId,
 		hash: jsonToFilenameSafeString(data)
 	});
-	let result = staticClientCache?.get(url);
-	result = await fetch(url, { method: "GET" }).then((r) => r.json()).then((d) => fromJSON(d, { plugins: getDefaultSerovalPlugins() }));
-	return result;
+  if (typeof document !== "undefined") {
+    let result = staticClientCache?.get(url);
+    result = await fetch(url, {
+      method: "GET"
+    }).then((r) => r.json()).then((d) => fromJSON(d, { plugins: getDefaultSerovalPlugins() }));
+    return result;
+  } else {
+    try {
+      return await fs.readFile(path.join('./public', url), 'utf-8').then(v => JSON.parse(v)).then((d) => fromJSON(d, { plugins: getDefaultSerovalPlugins() }));
+    } catch {
+      return null;
+    }
+  }
 };
 var staticFunctionMiddleware = createMiddleware({ type: "function" }).client(async (ctx) => {
-	if (process.env.NODE_ENV === "production" && typeof document !== "undefined") {
+	if (process.env.NODE_ENV === "production") {
 		const response = await fetchItem({
 			functionId: ctx.serverFnMeta.id,
 			data: ctx.data
