diff --git a/dist/esm/staticFunctionMiddleware.js b/dist/esm/staticFunctionMiddleware.js
index 3f9b7db31b55634034ac348ef8812b62aa0ff922..458489e3a381c2751227752d12abe6b9762e2e54 100644
--- a/dist/esm/staticFunctionMiddleware.js
+++ b/dist/esm/staticFunctionMiddleware.js
@@ -49,14 +49,22 @@ async function addItemToCache({ functionId, data, response }) {
 	await fs.writeFile(filePath, stringifiedResult, "utf-8");
 }
 var fetchItem = async ({ data, functionId }) => {
-	const url = await getStaticCacheUrl({
-		functionId,
-		hash: jsonToFilenameSafeString(data)
-	});
-	return fetch(url, { method: "GET" }).then((r) => r.json()).then((d) => fromJSON(d, { plugins: getSerovalPlugins(defaultSerovalDeserializerPlugins) }));
+  const url = await getStaticCacheUrl({
+    functionId,
+    hash: jsonToFilenameSafeString(data)
+  });
+  if (typeof document !== "undefined") {
+    return fetch(url, { method: "GET" }).then((r) => r.json()).then((d) => fromJSON(d, { plugins: getSerovalPlugins(defaultSerovalDeserializerPlugins) }));
+  } else {
+    try {
+      return await fs.readFile(path.join('./public', url), 'utf-8').then(v => JSON.parse(v)).then((d) => fromJSON(d, { plugins: getSerovalPlugins(defaultSerovalDeserializerPlugins) }));
+    } catch {
+      return null;
+    }
+  }
 };
 var staticFunctionMiddleware = createMiddleware({ type: "function" }).client(async (ctx) => {
-	if (process.env.NODE_ENV === "production" && typeof document !== "undefined") {
+	if (process.env.NODE_ENV === "production") {
 		const response = await fetchItem({
 			functionId: ctx.serverFnMeta.id,
 			data: ctx.data
